libpf/pfelf/file.go

```diff
@@ -882,3 +894,49 @@ func (f *File) DynString(tag elf.DynTag) ([]string, error) {
 func (f *File) IsGolang() bool {
 	return f.Section(".go.buildinfo") != nil || f.Section(".gopclntab") != nil
 }
+
+func (f *File) FilePath() (string, error) {
+	if f.InsideCore {
+		return "", errors.New("file path not available for ELF inside coredump")
+	}
+	return f.filePath, nil
+}
+
+// HasDWARFData is a copy of pfelf.HasDWARFData, but for the libpf.File interface.
+func (f *File) HasDWARFData() bool {
+	hasBuildID := false
+	hasDebugStr := false
+	for _, section := range f.Sections {
+		// NOBITS indicates that the section is actually empty, regardless of the size in the
+		// section header.
+		if section.Type == elf.SHT_NOBITS {
+			continue
+		}
+
+		if section.Name == buildIDSectionName {
+			hasBuildID = true
+		}
+
+		if slices.Contains(debugStrSectionNames, section.Name) {
+			hasDebugStr = section.Size > 0
+		}
+
+		// Some files have suspicious near-empty, partially stripped sections; consider them as not
+		// having DWARF data.
+		// The simplest binary gcc 10 can generate ("return 0") has >= 48 bytes for each section.
+		// Let's not worry about executables that may not verify this, as they would not be of
+		// interest to us.
+		if section.Size < 32 {
```

Comment:
good thinking

Reply:
to be fair all the credit goes to the original authors of the function https://github.com/DataDog/otel-profiling-agent/blob/967d1d580e39d8f63ca4dbf18b60f6755287a47a/libpf/pfelf/pfelf.go#L195 👌